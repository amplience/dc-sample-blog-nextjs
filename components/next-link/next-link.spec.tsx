/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';

import NextLink from './next-link';

const mockUseRouter = jest.fn();
jest.mock('next/router', () => {
  return {
    useRouter: () => mockUseRouter()
  };
});

describe('NextLink', () => {
  test('renders next link as an anchor tag', async () => {
    mockUseRouter.mockImplementationOnce(() => ({ route: '/', query: {} }));
    const component = await renderer.create(
      <NextLink href="/href-link/[..test]" as="/href-link/a-test">
        <div className="something-to-link"></div>
      </NextLink>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('renders next link as an anchor tag with aria-label', async () => {
    mockUseRouter.mockImplementationOnce(() => ({ route: '/', query: {} }));
    const component = await renderer.create(
      <NextLink href="/href-link/[..test]" as="/href-link/a-test" ariaLabel="link title">
        <div className="something-to-link"></div>
      </NextLink>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
