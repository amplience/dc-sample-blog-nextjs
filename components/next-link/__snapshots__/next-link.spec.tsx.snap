// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NextLink renders next link as an anchor tag 1`] = `
<a
  aria-label=""
  className="jsx-3574172415"
  href="/href-link/a-test"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="something-to-link"
  />
</a>
`;

exports[`NextLink renders next link as an anchor tag with aria-label 1`] = `
<a
  aria-label="link title"
  className="jsx-3574172415"
  href="/href-link/a-test"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  <div
    className="something-to-link"
  />
</a>
`;
